src/components/my-widgets-page.jsx

```diff
@@ -46,17 +46,22 @@ const MyWidgetsPage = () => {
 	const copyWidget = useCopyWidget()
 	const deleteWidget = useDeleteWidget()
 	readFromStorage()
-	const { data: widgets, isLoading, isFetching } = useQuery({
-		queryKey: 'widgets',
-		queryFn: apiGetWidgetInstances,
-		staleTime: Infinity
-	})
-	const { data: user} = useQuery({
+
+	const [page, setPage] = useState(1)
+	const [widgetsList, setWidgetsList] = useState([])
+	const {
+		data,
+		isLoading,
+		isFetching,
+		refetch,
+	} = useQuery('widgets', () => apiGetPaginatedWidgetInstances(page), { keepPreviousData: true, })
```

Comment:
One thing I'm noticing is that this query will continue to ping the server periodically with a page value above the `total_num_pages` value returned by the server response. Ideally, once all pages are fetched, the query no longer needs to run continually. There are a couple ways to disable the query, like adding a condition to the `enabled` option, or the `refetch` or `retry` options.

Reply:
I solved this by implementing `refetchOnWindowFocus: false` to the query options.